### PR TITLE
feat(inko): add queries for Inko

### PR DESCRIPTION
**Pull request by yorickpeterse**
This adds text object support for [Inko](https://inko-lang.org/).

I wasn't too sure which query file to use as a reference since they differ a bit, so I based the implementation on the queries file used for Rust.

### CODE REVIEW

File: queries/inko/textobjects.scm [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+    (#make-range! "class.inner" @_start @_end)))
+
+(reopen_class) @class.outer
+
+(reopen_class
+  body: (reopen_class_body
+    .
+    "{"
+    .
+    (_) @_start @_end
```
Comment (ribru17) [inline queries/inko/textobjects.scm:50]: Would you mind removing the `@_end` captures from the `@_start` captures? Sorry this is something we are going to remove across the repo soon

File: queries/inko/textobjects.scm [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+    (_)? @_end
+    .
+    "}"
+    (#make-range! "loop.inner" @_start @_end))) @loop.outer
+
+; Conditionals
```
Comment (ribru17) [inline queries/inko/textobjects.scm:93]: Would you mind adding something like
```
(while
  condition: (_) @conditional.inner)
```
(or an equivalent that you deem most appropriate)?

File: queries/inko/textobjects.scm [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+    "{"
+    .
+    (_) @_start @_end
+    (_)? @_end
+    .
+    "}"
+    (#make-range! "class.inner" @_start @_end)))
+
+; Methods
+(method) @function.outer
```
Comment (ribru17) [inline queries/inko/textobjects.scm:57]: Should these be given to `(closure)`?

**Comment by ribru17** [comment]
Thank you!

File: queries/inko/textobjects.scm [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+(method
+  body: (block
+    .
+    "{"
+    .
+    (_) @_start
+    (_)? @_end
+    .
+    "}"
+    (#make-range! "function.inner" @_start)))
```
Comment (ribru17) [inline queries/inko/textobjects.scm:68]: Looks like this directive and many (all?) others are missing the second parameter: should be (for this one)
```query
(#make-range! "function.inner" @_start @_end)
```

Comment (yorickpeterse) [reply queries/inko/textobjects.scm:68]: @ribru17 I removed this in response to your request in https://github.com/nvim-treesitter/nvim-treesitter-textobjects/pull/610#discussion_r1595773398. Did you mean to only remove `@_end` from certain nodes (and if so, which ones)?

Comment (ribru17) [reply queries/inko/textobjects.scm:68]: Sorry, I should have been clearer. Basically, the `@_end` should be removed in `(_) @_start @_end`, but not in the `#make-range!` predicate. That way the make range still uses the end capture, but makes sure it always uses the last end capture (not the one that was also applied to the start capture)

Comment (yorickpeterse) [reply queries/inko/textobjects.scm]: @ribru17 Ah gotcha, this should be fixed now :smiley:

Comment (ribru17) [reply queries/inko/textobjects.scm]: Thank you!

**Review by ribru17 — APPROVED** [review]

**Pull request merged**